tree/tree/test/friendinfo.cxx

```diff
@@ -50,17 +50,17 @@ void FillTree(const std::string &fileName, const std::string &treeName, int nEnt
    f.Close();
 }
 
-constexpr static std::size_t nEntriesInMainTree{20};
-constexpr static std::size_t nEntriesInFriendTree{5};
-const static std::string mainTreeName{"rfriendinfo_test_main"};
-const static std::string mainFileName{"rfriendinfo_test_main.root"};
-const static std::vector<std::string> friendTreeNames{
+constexpr std::size_t nEntriesInMainTree{20};
+constexpr std::size_t nEntriesInFriendTree{5};
+const std::string mainTreeName{"rfriendinfo_test_main"};
```

Comment:
Why does it recommend to remove the static?

Reply:
It's redundant, because these variables are in the anonymous namespace: https://clang.llvm.org/extra/clang-tidy/checks/readability/static-definition-in-anonymous-namespace.html

Note that these are not local variables, but global variables. So `static` has the effect of limiting the visibility to this translation unit, which the anonymous namespace does as well.

Reply:
> but global variables

I missed that :)

> So static has the effect of limiting the visibility to this translation unit

This may or may not be still a concern but it used to be that a difference between unnamed namespace and static was that the unnamed namespace was published a random/arbitrary symbol in the external symbol table while the static did not.  Hence static was reducing the resources cosumption/cost of the symbol.

Reply:
I don't think this is a concern here, it it were true what you say with the symbols. The file is only a test executable, so IMHO it doesn't really matter what the external symbols are.

Reply:
Fair enough.